### ISSUE TITLE
Fails when CommaDelimitedList is given to the argument of Fn::Join

### ISSUE THREAD

**Issue by komikoni** [mhlabs/cfn-diagram#15]
Using CommaDelimitedList as an argument of Fn :: Join in cfn results in an error

```yaml
  DeployBucketCrossReferencedAwsAccountIds:
    Type: CommaDelimitedList
    Default: ""

Conditions:
  ExistsDeployBucketCrossReferencedAwsAccountIds:
    !Not [
      !Equals [!Join [",", !Ref DeployBucketCrossReferencedAwsAccountIds], ""],
    ]
```

error message

```
$ cfn-dia draw.io -t template.yaml
(node:24640) UnhandledPromiseRejectionWarning: TypeError: current[k].Fn::Join[1].join is not a function
    at createPseudoResources (C:\Users\komik\AppData\Roaming\npm\node_modules\@mhlabs\cfn-diagram\resources\JsonUtil.js:7:46)
    at createPseudoResources (C:\Users\komik\AppData\Roaming\npm\node_modules\@mhlabs\cfn-diagram\resources\JsonUtil.js:10:7)
    at createPseudoResources (C:\Users\komik\AppData\Roaming\npm\node_modules\@mhlabs\cfn-diagram\resources\JsonUtil.js:10:7)
    at createPseudoResources (C:\Users\komik\AppData\Roaming\npm\node_modules\@mhlabs\cfn-diagram\resources\JsonUtil.js:10:7)
    at createPseudoResources (C:\Users\komik\AppData\Roaming\npm\node_modules\@mhlabs\cfn-diagram\resources\JsonUtil.js:10:7)
    at Object.createPseudoResources (C:\Users\komik\AppData\Roaming\npm\node_modules\@mhlabs\cfn-diagram\resources\JsonUtil.js:10:7)
    at generate (C:\Users\komik\AppData\Roaming\npm\node_modules\@mhlabs\cfn-diagram\index.js:89:12)
    at Command.<anonymous> (C:\Users\komik\AppData\Roaming\npm\node_modules\@mhlabs\cfn-diagram\index.js:44:11)
    at Command.listener [as _actionHandler] (C:\Users\komik\AppData\Roaming\npm\node_modules\@mhlabs\cfn-diagram\node_modules\commander\index.js:413:31)
    at Command._parseCommand (C:\Users\komik\AppData\Roaming\npm\node_modules\@mhlabs\cfn-diagram\node_modules\commander\index.js:914:14)
(node:24640) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:24640) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by komikoni** [comment]
full template.yaml

```yaml
AWSTemplateFormatVersion: 2010-09-09
Transform:
  - AWS::Serverless-2016-10-31
Description: |-
  APIサービスアーキテクチャのCI/CDシステム管理スタック

Parameters:
  ResourcePrefix:
    Type: String
    Description: |-
      リソース名を一意に表記するためのプレフィックス
      S3バケット名には大文字アルファベットが利用できないため、本パラメーターに大文字アルファベットは設定できません。
    ConstraintDescription: Resource Prefix must be between 2 and 20 characters, begin with a letter, and only contain lowercase letters, numbers, and hyphens (-).
    MinLength: 2
    MaxLength: 20
    AllowedPattern: ^[a-z][a-z0-9-]+$
    Default: api-service

  RepositoryName:
    Type: String
    Description: |-
      CodePipelineのソースコードとしてフックする対象リポジトリ名
      同一AWSアカウント内に存在するCodeCommit名を指定します。
    MinLength: 1
    MaxLength: 100
    Default: api-service-architecture

  CICDTemplateRepositoryPath:
    Type: String
    Description: |-
      対象CodeCommitリポジトリにおける、各ブランチのCI/CD環境構築用CloudFormationファイルのパスとファイル名
      パスとファイル名は、スラッシュ区切りで指定します。
      既定値は、API Service Architectureリポジトリのファイルを指定しています。
    Default: cicd/template-cicd.yaml

  CICDBuildspecRepositoryPath:
    Type: String
    Description: |-
      対象CodeCommitリポジトリにおける、CodeBuildで利用するbuildspecファイルのパスとファイル名
      各ブランチのbuildspecファイルのパスとファイル名は、スラッシュ区切りで指定します。
      既定値は、API Service Architectureリポジトリのファイルを指定しています。
    Default: cicd/buildspec.yaml

  DeployBucketCrossReferencedAwsAccountIds:
    Type: CommaDelimitedList
    Description: |-
      CI/CDで作成したデプロイ資材用のS3バケットについて (CICDCommonDeployBucket) 
      本番環境、ステージング環境など、クロスアカウント参照するAWSアカウントIDを
      ','（カンマ）区切りで入力することで、複数のAWSアカウントからの参照を可能にできます。
      例： '000000000000,111111111111'
    Default: ""

Conditions:
  ExistsDeployBucketCrossReferencedAwsAccountIds:
    !Not [
      !Equals [!Join [",", !Ref DeployBucketCrossReferencedAwsAccountIds], ""],
    ]

Resources:
  #----- CodeCommit -----#
  # CodeCommitリポジトリは新規作成しないため除去
  # CodeCommitRepo:
  #   Type: AWS::CodeCommit::Repository
  #   Description: CodePipelineを実行するためのCodeCommitリポジトリ
  #   Properties:
  #     RepositoryDescription: !Sub 'API Service Architecture CodeCommit Repository for ${ResourcePrefix}'
  #     RepositoryName: !Ref RepositoryName


  # 本スタックの作成・削除時に全ブランチのCICD環境を作成（または更新）・削除するトリガー(CloudFormation Custom Resource)
  CICDCommonCloudFormationTrigger:
    Type: Custom::CICDCommonCloudFormationTrigger
    Properties:
      ServiceToken: !GetAtt CICDCommonManagerFunction.Arn

  # CodeCommitの個別ブランチの作成・更新・削除時に、ブランチ用のCICD環境を作成（または更新）・削除するトリガー(CloudWatch Event)
  CICDCommonCodeCommitBranchTrigger:
    Type: AWS::Events::Rule
    Properties:
      Description: ブランチ作成・削除時に、CICD環境を作成・削除するトリガー
      Name: !Sub ${ResourcePrefix}-CICDCommonCodeCommitBranchTrigger
      State: ENABLED
      EventPattern:
        source:
          - aws.codecommit
        detail-type:
          - CodeCommit Repository State Change
        resources:
          - !Sub arn:${AWS::Partition}:codecommit:${AWS::Region}:${AWS::AccountId}:${RepositoryName}
        detail:
          event:
            - referenceCreated
            - referenceDeleted
            - referenceUpdated
          referenceType:
            - branch
      Targets:
        - Id: CICDCommonManagerFunction
          Arn: !GetAtt CICDCommonManagerFunction.Arn

  CICDCommonCodeCommitBranchTriggerPermission:
    Type: AWS::Lambda::Permission
    Properties:
      Action: lambda:InvokeFunction
      Principal: events.amazonaws.com
      FunctionName: !Ref CICDCommonManagerFunction
      SourceArn: !GetAtt CICDCommonCodeCommitBranchTrigger.Arn

  # CICDCommonManagerFunction更新時に、全ブランチのCICD環境を作成（または更新）するトリガー(CloudWatch Event)
  CICDCommonManagerFunctionUpdateTrigger:
    Type: AWS::Events::Rule
    Properties:
      Description: CICDCommonManagerFunctionの更新を検知するトリガー
      Name: !Sub ${ResourcePrefix}-CICDCommonManagerFunctionUpdateTrigger
      State: ENABLED
      EventPattern:
        source:
          - aws.lambda
        detail-type:
          - AWS API Call via CloudTrail
        detail:
          eventSource:
            - lambda.amazonaws.com
          eventName:
            # https://docs.aws.amazon.com/lambda/latest/dg/logging-using-cloudtrail.html 日付やバージョンがつくことがある
            # https://gist.github.com/tecmaverick/df31ce13a25be8e839961195d7cf53be 有志による情報
            # - UpdateFunctionCode
            # - UpdateFunctionConfiguration
            - UpdateFunctionCode20150331v2
            - UpdateFunctionConfiguration20150331v2
          requestParameters:
            functionName:
              # イベントにより、ファンクション名だけかARN名かが違うため、両方指定
              - !GetAtt CICDCommonManagerFunction.Arn  # UpdateFunctionCode20150331v2
              - !Ref CICDCommonManagerFunction         # UpdateFunctionConfiguration20150331v2
      Targets:
        - Id: CICDCommonManagerFunction
          Arn: !GetAtt CICDCommonManagerFunction.Arn

  CICDCommonManagerFunctionUpdateTriggerPermission:
    Type: AWS::Lambda::Permission
    Properties:
      Action: lambda:InvokeFunction
      Principal: events.amazonaws.com
      FunctionName: !Ref CICDCommonManagerFunction
      SourceArn: !GetAtt CICDCommonManagerFunctionUpdateTrigger.Arn

  CICDCommonManagerFunction:
    Type: AWS::Serverless::Function
    Properties:
      FunctionName: !Sub ${ResourcePrefix}-CICDCommonManagerFunction
      Handler: index.handler
      Runtime: nodejs12.x
      MemorySize: 1024
      Timeout: 300
      Role: !GetAtt CICDCommonRole.Arn
      Environment:
        Variables:
          RESOURCE_PREFIX: !Ref ResourcePrefix
          REPOSITORY_NAME: !Ref RepositoryName
          ROLE_ARN: !GetAtt CICDCommonRole.Arn
          BUILD_BUCKET: !Ref CICDCommonBuildBucket
          DEPLOY_BUCKET: !Ref CICDCommonDeployBucket
          CICD_TEMPLATE_REPOSITORY_PATH: !Ref CICDTemplateRepositoryPath
          CICD_BUILDSPEC_REPOSITORY_PATH: !Ref CICDBuildspecRepositoryPath
      CodeUri: cicd-manager-function/

  #----- IAM Role (for CI/CD) -----#
  # CI/CD環境構築全般に利用する実行ロール
  CICDCommonRole:
    Type: AWS::IAM::Role
    Properties:
      Description: |-
        IAM Role used for the CI/CD (CodePipeline, CodeBuild, CloudFormation, Cloudwatch Event, Lambda)
      RoleName: !Sub ${ResourcePrefix}-CICDCommonRole
      AssumeRolePolicyDocument:
        Statement:
          - Sid: CICDAssumeRole
            Effect: Allow
            Action: sts:AssumeRole
            Principal:
              Service:
                - codepipeline.amazonaws.com
                - codebuild.amazonaws.com
                - cloudformation.amazonaws.com
                - events.amazonaws.com
                - lambda.amazonaws.com
      Policies:
        - PolicyName: !Sub ${ResourcePrefix}-CICDCommonRolePolicy
          PolicyDocument:
            Id: !Sub ${ResourcePrefix}-ci-cd-role-policy
            Version: 2012-10-17
            Statement:
              - Sid: DeployResouecesExecution
                # CI/CD実行時に必要なIAMポリシー
                Effect: Allow
                Action:
                  - events:*
                  - codecommit:*
                  - codepipeline:*
                  - codebuild:*
                  - cloudformation:*
                  - s3:*
                  - logs:*
                  - apigateway:*
                  - cognito-idp:*
                  - lambda:*
                  - dynamodb:*
                  # TODO: IAMについては要見直し
                  - iam:*Role
                  - iam:*RolePolicy
                  - iam:Tag*
                  - iam:Untag*
                Resource: "*"

  #----- S3 -----#
  # ビルド処理の一時的なデータ（アーティファクト）の引継に利用するS3バケット
  CICDCommonBuildBucket:
    Type: AWS::S3::Bucket
    Properties:
      BucketName: !Sub ${ResourcePrefix}-cicd-common-build
      BucketEncryption:
        ServerSideEncryptionConfiguration:
          - ServerSideEncryptionByDefault:
              SSEAlgorithm: AES256
      # 1日経過したら削除
      LifecycleConfiguration:
        Rules:
          - Id: !Sub ${ResourcePrefix}-cicd-common-build-LifeCycle
            Status: Enabled
            ExpirationInDays: 1
            AbortIncompleteMultipartUpload:
              DaysAfterInitiation: 1

  # CI/CD関連処理以外によるオブジェクト変更を不可とするためのバケットポリシー
  CICDCommonBuildBucketPolicy:
    Type: AWS::S3::BucketPolicy
    Properties:
      Bucket: !Ref CICDCommonBuildBucket
      PolicyDocument:
        Id: !Sub ${ResourcePrefix}-cicd-common-build-bucket-policy
        Version: 2012-10-17
        Statement:
          - Sid: DefaultDenyPutWithoutCICDCommonRole
            # CICDCommonRoleを利用するリソース以外は書換え不可にする
            Effect: Deny
            Condition:
              StringNotLike:
                aws:userid: !Sub ${CICDCommonRole.RoleId}:*
            Action:
              # - s3:Put* # TODO: 要見直し：CloudFormationでバケットポリシーの更新(s3:PutBucketPolicy)に失敗する為、
              #                             PutObject関連のみ拒否するよう変更
              - s3:PutObject
              - s3:PutObject*
            Resource:
              - !Sub arn:${AWS::Partition}:s3:::${CICDCommonBuildBucket}
              - !Sub arn:${AWS::Partition}:s3:::${CICDCommonBuildBucket}/*
            Principal: "*"

  # デプロイリソースを保管・参照する為のバケット
  # デプロイリソースには、SAM(CloudFormation)テンプレート,Lambda Zip,OAS等を含む
  CICDCommonDeployBucket:
    Type: AWS::S3::Bucket
    DeletionPolicy: Retain
    UpdateReplacePolicy: Retain # [cfn-lint]W3011対応
    Properties:
      BucketName: !Sub ${ResourcePrefix}-cicd-common-deploy
      BucketEncryption:
        ServerSideEncryptionConfiguration:
          - ServerSideEncryptionByDefault:
              SSEAlgorithm: AES256
      VersioningConfiguration:
        Status: Enabled

  # CI/CD以外によるオブジェクト変更を不可とするためのバケットポリシー
  # また、クロスアカウント(ステージング、本番等)へのデプロイ用に読取アクセス許可を付与
  # スタック削除時はリソースを残存させる
  CICDCommonDeployBucketPolicy:
    Type: AWS::S3::BucketPolicy
    DeletionPolicy: Retain
    UpdateReplacePolicy: Retain # [cfn-lint]W3011対応
    Properties:
      Bucket: !Ref CICDCommonDeployBucket
      PolicyDocument:
        Id: !Sub ${ResourcePrefix}-cicd-common-deploy-bucket-policy
        Version: 2012-10-17
        Statement:
          - Sid: DefaultDenyPutWithoutCICDCommonRole
            # CICDCommonRoleを持つリソース（本スタックのリソース）以外は書換え不可
            Effect: Deny
            Condition:
              StringNotLike:
                aws:userid: !Sub ${CICDCommonRole.RoleId}:*
            Action:
              # - s3:Put* # TODO: 要見直し：CloudFormationでバケットポリシーの更新(s3:PutBucketPolicy)に失敗する為、
              #                             PutObject関連のみ拒否するよう変更
              - s3:PutObject
              - s3:PutObject*
            Resource:
              - !Sub arn:${AWS::Partition}:s3:::${CICDCommonDeployBucket}
              - !Sub arn:${AWS::Partition}:s3:::${CICDCommonDeployBucket}/*
            Principal: "*"
          # クロスアカウント・オブジェクトアクション用ポリシー（読み取り専用）
          - !If
            - ExistsDeployBucketCrossReferencedAwsAccountIds
            - Sid: CrossAccountAccessforGetAction
              Effect: Allow
              Action: s3:Get*
              Resource: !Sub arn:${AWS::Partition}:s3:::${CICDCommonDeployBucket}/${ResourcePrefix}-master/*
              Principal:
                AWS: !Ref DeployBucketCrossReferencedAwsAccountIds
            - !Ref AWS::NoValue
          # クロスアカウント・バケットアクション用ポリシー（読み取り専用）
          - !If
            - ExistsDeployBucketCrossReferencedAwsAccountIds
            - Sid: CrossAccountAccessforListAction
              Effect: Allow
              Action: s3:List*
              Resource: !Sub arn:${AWS::Partition}:s3:::${CICDCommonDeployBucket}
              Condition:
                StringLike:
                  s3:prefix: !Sub ${ResourcePrefix}-master/*
              Principal:
                AWS: !Ref DeployBucketCrossReferencedAwsAccountIds
            - !Ref AWS::NoValue
Outputs:
  AllCloudFormationListURL:
    Value: !Sub https://${AWS::Region}.console.aws.amazon.com/cloudformation/home?region=${AWS::Region}#/stacks?filteringText=${ResourcePrefix}
  AllCodePipelineListURL:
    Value: !Sub https://${AWS::Region}.console.aws.amazon.com/codesuite/codepipeline/pipelines?region=${AWS::Region}&pipelines-meta=%7B%22f%22%3A%7B%22text%22%3A%22${ResourcePrefix}%22%7D%2C%22s%22%3A%7B%22property%22%3A%22updated%22%2C%22direction%22%3A-1%7D%2C%22n%22%3A10%2C%22i%22%3A0%7D
  CICDCommonManagerFunctionCloudWatchLogsURL:
    Value: !Sub https://${AWS::Region}.console.aws.amazon.com/cloudwatch/home?region=${AWS::Region}#logsV2:log-groups/log-group/$252Faws$252Flambda$252F${CICDCommonManagerFunction}
  MasterBrachDeployBucketS3URL:
    Value: !Sub https://console.aws.amazon.com/s3/buckets/${CICDCommonDeployBucket}/${ResourcePrefix}-master/?region=${AWS::Region}

```

**Comment by komikoni** [comment]
This issue has been resolved in v1.1.9.

```
$ cfn-dia -v
1.1.9
$ cfn-dia draw.io -t template.yaml
Writing diagram to ./template.drawio
Press CTRL+C to exit
? Options (Use arrow keys)
> Filter resources by type
  Filter resources by name
  Edge labels: On
```
